pkg/reconciler/instances/istio/tests/integration_test.go

```diff
@@ -0,0 +1,167 @@
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/avast/retry-go"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/kubectl/pkg/util/podutils"
+)
+
+const (
+	istioNamespace         = "istio-system"
+	pilotIngressGwSelector = "istio in (pilot, ingressgateway)"
+	healthzTestURLFormat   = "http://%s:%s/healthz/ready"
+	gatewayManifest        = `
```

Comment:
How about extracting these resources to dedicated files under some `resources/` dir? Maybe that would allow easier maintenance or adjustments in the future